account_analytic_salesperson/models/sale_order.py

```diff
@@ -0,0 +1,23 @@
+# Copyright 2019 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from odoo import api, models
+
+
+class SaleOrder(models.Model):
+
+    _inherit = 'sale.order'
+
+    @api.multi
+    def _create_analytic_account(self, prefix=None):
+        """Propagate sale order salesperson through context key"""
+        self = self.with_context(analytic_user_id=self.user_id.id)
+        return super()._create_analytic_account(prefix=prefix)
+
+    @api.multi
+    def action_confirm(self):
+        """Ensure analytic account is created after sale order confirmation"""
+        res = super().action_confirm()
+        for order in self:
+            if not order.analytic_account_id:
```

Comment:
IMHO, even if you mention in DESCRIPTION that you ensure analytic account is created for each sale order, it is a change in the standard behaviour. Is it really the needs ?

You use the expense product here https://github.com/OCA/account-analytic/pull/252/files#diff-938c8d23d854c68b16b71b3ec1bcd046R12

So, the Odoo stnadard behaviour will apply, no ?

Reply:
@rousseldenis Yes, it is intended (as stated in readme). 
I'll add a note in ROADMAP, that it could probably be split into two modules but at the moment I don't have time to rework it that much.

The expense product is only used in the test to make sure behaviour doesn't change :wink: